terraform/modules/daos_server/scripts/client_install.sh

```diff
@@ -12,61 +13,79 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-# Install DAOS Client package
 #
-# TODO: Add support for installing on openSUSE Leap 15.3 and Ubuntu 20.04 LTS
+# Install DAOS Client package
 #
+DAOS_VERSION="${DAOS_VERSION:-2.2}"
 
-DAOS_VERSION="${DAOS_VERSION:-2.0}"
-
-echo "BEGIN: DAOS Client Installation"
-
-# Determine which repo to use
-# shellcheck disable=SC1091
-source "/etc/os-release"
-OS_VERSION=$(echo "${VERSION_ID}" | cut -d. -f1)
-OS_VERSION_ID="${ID,,}_${OS_VERSION}"
-case ${OS_VERSION_ID} in
-  centos_7)
-    DAOS_OS_VERSION="CentOS7"
-    ;;
-  centos_8)
-    DAOS_OS_VERSION="CentOS8"
-    ;;
-  rocky_8)
-    DAOS_OS_VERSION="CentOS8"
-    ;;
-  *)
-    echo "ERROR: Unsupported OS: ${OS_VERSION_ID}. Exiting."
-    exit 1
-    ;;
-esac
+set_vars() {
+  # shellcheck disable=SC1091
+  source "/etc/os-release"
+  OS_VERSION_ID="${ID,,}_${VERSION_ID}"
+  OS_MAJOR_VERSION=$(echo "${VERSION_ID}" | cut -d. -f1)
+  OS_MAJOR_VERSION_ID="${ID,,}_${OS_MAJOR_VERSION}"
 
-if ! rpm -qa | grep -q "epel-release"; then
-  yum install -y epel-release
-fi
+  case "${OS_MAJOR_VERSION_ID}" in
+    centos_7)
+      DAOS_OS_VERSION="CentOS7"
+      PKG_MGR="yum"
+      REPO_PATH=/etc/yum.repos.d
+      ;;
+    almalinux_8|centos_8|rhel_8|rocky_8)
+      DAOS_OS_VERSION="EL8"
+      PKG_MGR="dnf"
+      REPO_PATH=/etc/yum.repos.d
+      ;;
+    opensuse-leap_15)
+      if [[ "${OS_VERSION_ID}" == "opensuse-leap_15.4" ]]; then
+        log.error "Unsupported OS: ${OS_VERSION_ID}."
+        log.error "See https://daosio.atlassian.net/browse/DAOS-11637"
+        exit 1
+      fi
+      DAOS_OS_VERSION="Leap15"
+      PKG_MGR="zypper"
+      REPO_PATH=/etc/zypp/repos.d
+      ;;
+    *)
+      log.error "Unsupported OS: ${OS_VERSION_ID}. Exiting."
+      exit 1
+      ;;
+  esac
+}
 
-if [ ! -f $(which wget) ];then
-  yum -y install wget
-fi
+install_epel() {
+  # DAOS has dependencies on packages in epel
+  echo "Adding epel-release repo"
+  if [[ "${OS_MAJOR_VERSION_ID}" != "opensuse-leap_15" ]]; then
+    if ! rpm -qa | grep -q "epel-release"; then
+      $PKG_MGR install -y "https://dl.fedoraproject.org/pub/epel/epel-release-latest-${OS_MAJOR_VERSION}.noarch.rpm"
```

Comment:
Seems strange to use fedora repo with opensuse

Reply:
Fixed to ensure epel-release is only installed if the OS is not opensuse (any version).